internal/server/server.go

```
@@ -0,0 +1,429 @@
+
+func NewServer() *Server {
+	// Set Gin mode based on environment
+	if os.Getenv("GIN_MODE") == "" {
+		gin.SetMode(gin.ReleaseMode)
+	}
+
+	router := gin.Default()
+
+	// Configure trusted proxies based on environment
+	if proxyList := os.Getenv("TRUSTED_PROXIES"); proxyList != "" {
+		router.SetTrustedProxies(strings.Split(proxyList, ","))
```

golangci-lint (1.62.2)

Error return value of `router.SetTrustedProxies` is not checked

(errcheck)